content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx

```diff
@@ -0,0 +1,530 @@
+---
+title: 'Composite types'
+metaTitle: 'Composite types'
+metaDescription: 'Composite types'
+tocDepth: 2
+preview: true
+---
+
+<TopBlock>
+
+<Admonition type="info">
+
+Composite types are available in Preview in versions `3.10.0` and later.
+
+</Admonition>
+
+<Admonition type="warning">
+
+Composite types are currently only available on MongoDB.
+
+</Admonition>
+
+[Composite types](/concepts/components/prisma-schema/data-model#defining-composite-types), known as [embedded documents](https://docs.mongodb.com/manual/core/data-model-design/#std-label-data-modeling-embedding) in MongoDB, allow you to embed records within other records.
+
+This page explains how to:
+
+- [create](#creating-composite-types-with-create-and-createmany) new records with composite types using `create` and `createMany`
+- [change](#changing-composite-types-within-update-and-updatemany) the composite types within existing records using `update` and `updateMany`
+- [delete](#deleting-composite-types-with-delete-and-deletemany) records with composite types using `delete` and `deleteMany`
+- [filter](#filtering-composite-types-with-find-and-findmany) records with composite types using `find` and `findMany`
+
+</TopBlock>
+
+## Example schema
+
+We’ll use this schema for the examples that follow:
+
+```prisma file=schema.prisma
+model Product {
+  id     String  @id
+  name   String
+  price  Decimal
+  colors Color[]
+  sizes  Size[]
+  photos Photo[]
+}
+
+model Order {
+  id              String   @id
+  product         Product
+  color           Color
+  size            Size
+  shippingAddress Address
+  billingAddress  Address?
+}
+
+enum Color {
+  Red
+  Green
+  Blue
+}
+
+enum Size {
+	Small
+	Medium
+	Large
+  XLarge
+}
+
+type Photo {
+  height Int
+  width  Int
+  url    String
+}
+
+type Address {
+  street String
+  city   String
+  zip    String
+}
+```
+
+In this schema, the `Product` model has a `Photo[]` composite type, and the `Order` model has two embedded `Address` types. The `shippingAddress` is required, but the `billingAddress` is optional.
+
+## Considerations when using composite types
+
+There are currently some limitations when using composite types in the Prisma Client:
+
+- [`findUnique`](/reference/api-reference/prisma-client-reference#findunique) can't filter on composite types
+- [`aggregrate`](/concepts/components/prisma-client/aggregation-grouping-summarizing#aggregate), [`groupBy`](/concepts/components/prisma-client/aggregation-grouping-summarizing#group-by), [`count`](/concepts/components/prisma-client/aggregation-grouping-summarizing#count) don’t support embedded operations.
```

Comment:
Seems like since we're going with composite types, we should also go with composite operations.

```suggestion
- [`aggregrate`](/concepts/components/prisma-client/aggregation-grouping-summarizing#aggregate), [`groupBy`](/concepts/components/prisma-client/aggregation-grouping-summarizing#group-by), [`count`](/concepts/components/prisma-client/aggregation-grouping-summarizing#count) don’t support composite operations.
```